### PR TITLE
Initial pytorch experiments

### PR DESCRIPTION
**Pull request by subutai**

### CODE REVIEW

File: htmresearch/frameworks/pytorch/resnet_models.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Modified from torchvision
```
Comment (lscheinkman) [inline htmresearch/frameworks/pytorch/resnet_models.py]: It looks like there were minor modifications to `ResNet` class, the rest of the code is identical. Why not just create a class that  inherits `ResNet`? 
You could update the `avgpool.kernel_size` and add the `learningIterations` in the subclass leaving the rest of this code in torchvision.

File: htmresearch/frameworks/pytorch/resnet_models.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                               padding=3,
+                               bias=False)
+        self.bn1 = nn.BatchNorm2d(64)
+        self.relu = nn.ReLU(inplace=True)
+        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+        self.layer1 = self._make_layer(block, 64, layers[0])
+        self.layer2 = self._make_layer(block, 128, layers[1], stride=2)
+        self.layer3 = self._make_layer(block, 256, layers[2], stride=2)
+        self.layer4 = self._make_layer(block, 512, layers[3], stride=2)
+        self.avgpool = nn.AvgPool2d(1, stride=1)
```
Comment (lscheinkman) [inline htmresearch/frameworks/pytorch/resnet_models.py]: Why not `7` like `torchvision`?

File: htmresearch/frameworks/pytorch/audio_transforms.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+# ----------------------------------------------------------------------
+
+"""
+Adapted from https://github.com/tugstugi/pytorch-speech-commands
+Google speech commands dataset.
+"""
+
+import random
+import numpy as np
+import librosa
```
[the remark this answers is not in the document]
Comment (subutai) [reply htmresearch/frameworks/pytorch/audio_transforms.py]: Good point

File: htmresearch/frameworks/pytorch/audio_transforms.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+        return data
+
+class ChangeAmplitude(object):
+    """Changes amplitude of an audio randomly."""
+
+    def __init__(self, amplitude_range=(0.7, 1.1)):
+        self.amplitude_range = amplitude_range
+
+    def __call__(self, data):
+        if not should_apply_transform():
```
[the remark this answers is not in the document]
Comment (subutai) [reply htmresearch/frameworks/pytorch/audio_transforms.py]: Yes, this is taken from the other repo. I haven't looked into how much of these we really need. Will leave it as-is and investigate later.

File: htmresearch/frameworks/pytorch/resnet_models.py (hunk 1 above)
Comment (subutai) [reply htmresearch/frameworks/pytorch/resnet_models.py]: Good point!

File: htmresearch/frameworks/pytorch/resnet_models.py (hunk 2 above)
Comment (subutai) [reply htmresearch/frameworks/pytorch/resnet_models.py]: No idea!

File: htmresearch/frameworks/pytorch/resnet_models.py (hunk 1 above)
Comment (subutai) [reply htmresearch/frameworks/pytorch/resnet_models.py]: Just did a quick version of this now, and removed the other classes that were unchanged.

**Pull request merged**